lib/miq_automation_engine/service_models/mixins/miq_ae_service_miq_request_mixin.rb

```diff
@@ -17,8 +17,10 @@ def user_message=(msg)
 
   def set_option(key, value)
     ar_method do
-      @object.options[key] = value
-      @object.update_attribute(:options, @object.options)
+      @object.with_lock do
```

Comment:
We observed that `with_lock` automatically does a reload. It is confirmed in this discussion too https://stackoverflow.com/questions/15694326/rails-activerecord-pessimistic-locking-do-i-need-to-reload-after-obtaining-a-l